### ISSUE TITLE
C# Calling `GD.Load<CustRes>(...)` from constructor causes invalid cast exception.

### ISSUE THREAD

**Issue by cgbeutler** [godotengine/godot#52729]
### Godot version

3.3.3

### System information

Windows 10

### Issue description

In C#, when calling `GD.Load<CustomType>(path)` from within a constructor causes an invalid cast exception that can cause the editor to crash.
The editor usually runs default constructors to build a temporary, defaulted monoobject. I suspect that this operation occurs before `Load<>` is fully populated with cast possibilities. As a result, calling `GD.Load<CustomType>(path)` will cause the editor to crash. The same can occur when calling load in a static context.

I guess my expectation is that Load would work in the default constructor and in static constexts.

### Steps to reproduce

Create a custom resource. (Can just be empty for this):
```csharp
// CustRes.cs
public class CustRes : Resource { }
```

Create another script of any type and add a load call in the default constructor:
```csharp
// LoadInConstructor.cs
public class LoadInConstructor : Node
{
    public LoadInConstructor()
    {
        var tmp = GD.Load<CustRes>("res://CustRes.tres");
    }
}
```

Now hit build and you should get an invalid cast exception thrown by the load call.

### Minimal reproduction project

[godot_error_ConstructorLoadCustomRes.zip](https://github.com/godotengine/godot/files/7173713/godot_error_ConstructorLoadCustomRes.zip)

**Comment by paulloz** [comment]
This is basically a duplicate of #36395. You'll see that tagging both classes with `[Tool]` will make this work.

**Comment by Calinou** [comment]
Closing per @paulloz's comment.

**Comment by cgbeutler** [comment]
@paulloz Hmmm, that's interesting. That bug is unfortunately named, but does indeed seem related. (Note for future folks, adding `[Tool]` is probly not a great workaround, as that is still a work in progress and even warned about on the main c# page.)

I will head over to that bug and see if we can get it rewritten to the root problem.

**Comment by paulloz** [comment]
I'm not saying it is a workaround, I am saying it is the same issue.  
There is nothing here about "populating cast possibilities" or something happening too early. It is a matter of that script not being available for instancing. So the object returned by `Load()` isn't castable to `CustomType` type because it is instanced as a higher class (`Resource` here).

**Comment by cgbeutler** [comment]
@paulloz , yeah, just clarifying for newer folks that that wasn't a workaround. I got chu.
And, yeah, my phrasing of "populating cast possibilities" was bad. I really just meant the scripts are not built for instancing in tool mode. I think I was trying to dumb it down.
